src/privatesend/privatesend-client.cpp

```diff
@@ -1650,10 +1650,17 @@ bool CPrivateSendClientSession::CreateDenominated(CAmount nBalanceToDenominate,
 
         auto countPossibleOutputs = [&](CAmount nAmount) -> int {
             std::vector<CAmount> vecOutputs;
-            while (txBuilder.TryAddOutputs(vecOutputs)) {
+            while(true) {
+                // Create an potential output
                 vecOutputs.push_back(nAmount);
+                if (!txBuilder.TryAddOutputs(vecOutputs) || txBuilder.CountOutputs() + vecOutputs.size() >= PRIVATESEND_DENOM_OUTPUTS_THRESHOLD) {
```

Comment:
```suggestion
                if (!txBuilder.TryAddOutputs(vecOutputs) || txBuilder.CountOutputs() + vecOutputs.size() > PRIVATESEND_DENOM_OUTPUTS_THRESHOLD) {
```
because `pop_back` below would result in `PRIVATESEND_DENOM_OUTPUTS_THRESHOLD - 1` max otherwise